python_scripts/functions.py

```diff
@@ -407,28 +410,38 @@ def generate_iam_config(
     iam_policy_path: str
         Optional path to output the iam policy json generated from the iam_config just generated
     """
+    if os.path.exists(iam_config_output) and overwrite_config is not True:
+        raise ValueError(
+            f"{iam_config_output} exists: to overwrite set overwrite_config=True"
+        )
+
+    log_path = config["log-base-path"].replace("s3://", "")
+    land_path = config["land-base-path"].replace("s3://", "")
+    fail_path = config["fail-base-path"].replace("s3://", "")
+    pass_path = config["pass-base-path"].replace("s3://", "")
```

Comment:
Some of these might be blank? I think `fail_path` is optional?

Reply:
Good point too, will add some trys/ifs in there

Reply:
have set it so you have to explicitly set fail path to `null`, as this appears to be in line with the rest of the code. The rest of the paths are required.